sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -871,6 +871,20 @@ same command, targeting the same device. There is the additional limitation that
 to update an executable graph, every node in the graph must be either a kernel
 command or a host-task.
 
+The only characteristics that can differ between two topologically identical
+graphs during an update are:
+
+* The arguments to kernel nodes may be different between graphs. For example,
+  the graph may capture different values for the USM pointers or accessors used
+  in the graph. It is these kernels arguments in `graph` that constitute the
+  inputs & outputs to update to.
+
+* The code that is executed in host-task or device kernel nodes may be
+  different between graphs. However, the update will only reflect the new code
+  in `graph` host-tasks nodes, as it doesn't need device-side interaction. Any
+  changes to executable code in `graph` kernel nodes will be ignored by the
+  update.
+
```

Comment:
Hmm, I wonder if we could detect differences in these nodes and error in that case rather than saying it's ignored.

If we don't want to commit to detecting it I wonder if saying the behaviour is undefined is better than saying it's ignored.

Reply:
Hard to say until we've implemented it, will update to say UB for now

Reply:
I actually changed my mind when writing this up and pushed a commit making it an error. Since from a users perspective that's the most informative behaviour, and we're already not implementing all the errors immediately, so don't need short-term wiggle room of "UB" for our implementation timeline.